src/features/calendar/components/index.tsx

```diff
@@ -55,17 +55,19 @@ const Calendar = () => {
   );
 
   useEffect(() => {
-    setFocusDate(getDateFromString(focusDateStr));
-  }, [focusDateStr]);
-
-  useEffect(() => {
-    setSelectedTimeScale(getTimeScale(timeScaleStr));
-  }, [timeScaleStr]);
```

Comment:
I just tried it now that you've removed this, and the browser back/forward keys are no longer working. Do you need to remove these for the flickering to stop? Otherwise, please bring them back.

Reply:
No it is not related with flickering error! I will write it back!